File: CS/WindowsApplication1/SerializableFormatCollection.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Collections;
using DevExpress.XtraTreeList.StyleFormatConditions;
using DevExpress.Utils;
using DevExpress.XtraGrid;
using DevExpress.Utils.Serializing;

namespace WindowsApplication1
{
    public class SerializableFormatCollection : CollectionBase
    {
        public SerializableFormatCollection(StyleFormatConditionCollection collection)
        {
            CopyFromStyleFormats(collection);
        }

        public SerializableFormatCollection()
        {
        }


        public SerializableFormatCondition Add()
        {
            SerializableFormatCondition serializableFormat = new SerializableFormatCondition();
            InnerList.Add(serializableFormat);
            return serializableFormat;
        }

        public void CopyFromStyleFormats(StyleFormatConditionCollection styles)
        {
            Clear();
            foreach (StyleFormatCondition style in styles)
            {
                SerializableFormatCondition serializableFormat = new SerializableFormatCondition();
                serializableFormat.CopyFromStyleFormat(style);
                InnerList.Add(serializableFormat);
            }
        }

        public void CopyToStyleFormats(StyleFormatConditionCollection styles)
        {
            styles.Clear();
            foreach (SerializableFormatCondition serializableFormat in this)
            {
                StyleFormatCondition style = new StyleFormatCondition();
                styles.Add(style);
                serializableFormat.CopyToStyleFormat(style);
            }
        }
    }
}

File: VB/WindowsApplication1/SerializableFormatCollection.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.Collections
Imports DevExpress.XtraTreeList.StyleFormatConditions
Imports DevExpress.Utils
Imports DevExpress.XtraGrid
Imports DevExpress.Utils.Serializing

Namespace WindowsApplication1
	Public Class SerializableFormatCollection
		Inherits CollectionBase
		Public Sub New(ByVal collection As StyleFormatConditionCollection)
			CopyFromStyleFormats(collection)
		End Sub

		Public Sub New()
		End Sub


		Public Function Add() As SerializableFormatCondition
			Dim serializableFormat As New SerializableFormatCondition()
			InnerList.Add(serializableFormat)
			Return serializableFormat
		End Function

		Public Sub CopyFromStyleFormats(ByVal styles As StyleFormatConditionCollection)
			Clear()
			For Each style As StyleFormatCondition In styles
				Dim serializableFormat As New SerializableFormatCondition()
				serializableFormat.CopyFromStyleFormat(style)
				InnerList.Add(serializableFormat)
			Next style
		End Sub

		Public Sub CopyToStyleFormats(ByVal styles As StyleFormatConditionCollection)
			styles.Clear()
			For Each serializableFormat As SerializableFormatCondition In Me
				Dim style As New StyleFormatCondition()
				styles.Add(style)
				serializableFormat.CopyToStyleFormat(style)
			Next serializableFormat
		End Sub
	End Class
End Namespace
